stl/inc/tuple

```diff
@@ -419,10 +419,27 @@ public:
         : tuple(_Unpack_tuple_t{}, _STD move(_Right)) {}
 #endif // ^^^ !_HAS_CONDITIONAL_EXPLICIT ^^^
 
+#if _HAS_CONDITIONAL_EXPLICIT
     template <class _Alloc, class _This2 = _This,
         enable_if_t<conjunction_v<is_default_constructible<_This2>, is_default_constructible<_Rest>...>, int> = 0>
+    explicit(
+        !conjunction_v<_Is_implicitly_default_constructible<_This2>, _Is_implicitly_default_constructible<_Rest>...>)
+        tuple(allocator_arg_t, const _Alloc& _Al)
+        : _Mybase(allocator_arg, _Al), _Myfirst(_Al, allocator_arg) {}
+#else // ^^^ _HAS_CONDITIONAL_EXPLICIT ^^^ / vvv !_HAS_CONDITIONAL_EXPLICIT vvv
+    template <class _Alloc, class _This2 = _This,
+        enable_if_t<
+            conjunction_v<_Is_implicitly_default_constructible<_This2>, _Is_implicitly_default_constructible<_Rest>...>,
+            int> = 0>
     tuple(allocator_arg_t, const _Alloc& _Al) : _Mybase(allocator_arg, _Al), _Myfirst(_Al, allocator_arg) {}
 
+    template <class _Alloc, class _This2 = _This,
+        enable_if_t<!conjunction_v<_Is_implicitly_default_constructible<_This2>,
+                        _Is_implicitly_default_constructible<_Rest>...>,
+            int> = 0>
+    explicit tuple(allocator_arg_t, const _Alloc& _Al) : _Mybase(allocator_arg, _Al), _Myfirst(_Al, allocator_arg) {}

```

Comment:
Each constraint also needs to mention the basic constraint of `conjunction_v<is_default_constructible<_This2>, is_default_constructible<_Rest>...>`. (For the first one, you can just use a single `conjunction_v`. For the second one, you'll need a `conjunction_v` and a `negation<conjunction` within. You should be able to use the constraints on the ordinary default constructor, unchanged.)

For example, suppose that the `Types` aren't constructible from `()` or `= {}` at all. As written, your `explicit` constructor would be enabled, and yet would fail to compile if instantiated. The idea behind all of these constraints is that `is_constructible` on the `tuple` will accurately report whether that constructor can actually be instantiated.

Reply:
Oh yes! I completely left out the initial constraint on the constructor, thanks!